apps/req.c

```diff
@@ -468,6 +468,15 @@ int req_main(int argc, char **argv)
             break;
         case OPT_MD:
             digest = opt_unknown();
+            {
+                EVP_MD *md = NULL;
+                if (!opt_md_silent(digest, &md)) {
```

Comment:
Add an empty line above.

Reply:
This will not work, if the user wants uses "load provider" options, see https://github.com/openssl/openssl/pull/15340#discussion_r635387894

Reply:
This will not work if the -provider is after the digest opt, yes. But there is not much you can do with it as there might be multiple "unknown" options used. You'd have to store them all and then check them. IMO not worth the hassle.

Reply:
>might be multiple "unknown" options used.

There can be only one; see https://github.com/openssl/openssl/blob/master/apps/lib/opt.c#L207 and not that it is outside the NDEBUG ifdef.  Now, yeah, the user might do "-md5 -sha256" duplicating it, and the code should check against that.

Reply:
I do check for the digest if nullptr for more digest options. [Look at it](https://github.com/openssl/openssl/pull/15362/files)